src/resources/views/crud/fields/relationship/select.blade.php

```diff
@@ -0,0 +1,234 @@
+@php
+
+    //in case entity is superNews we want the url friendly super-news
+
+    $connected_entity = new $field['model'];
+    $connected_entity_key_name = $connected_entity->getKeyName();
+
+    //we need to explicitly define field type to load correct assets
+    //this is needed because the field type `relationship` is a switch to
+    //other field types like select2
+    $field['type'] = 'select2';
```

Comment:
Same thing here - I think this should be moved to wherever the switchboard logic is. Since it only applies to that use case, it should be solved there, not here.